### PR TITLE
if minimum_version did not ignore the character '@' 

### PR DESCRIPTION
**Pull request by basvandervlies**
Changelog: if minimum_version did not ignore the character '@' (Redmne #7862)

### CODE REVIEW

File: libpromises/cf3lex.l [hunk 1]
```diff
@@ -252,11 +258,11 @@ promise_type   [a-zA-Z_]+:
...
-                               }
-
-<if_ignore_state>[^@]      {
-                               ParserDebug("\tL:inside macro @if, ignoring text:%s\n", yytext);
-                           }
+                               }\
+<if_ignore_state>.    {
+                          /* eat up al unknown chars when line starts with @*/
+                          ParserDebug("\tL:inside macro @if, ignoring text:%s\n", yytext);
+                      }
```
Comment (kacf) [inline libpromises/cf3lex.l]: @tzz: You recently made a change to add this '[^@]'. Will it still work with Bas' change?

File: tests/acceptance/00_basics/macros/if_ignore_with_addsign.cf [hunk 2]
```diff
@@ -0,0 +1,30 @@
+######################################################
+#
+#  Test that @if works for greater versions
```
Comment (kacf) [inline tests/acceptance/00_basics/macros/if_ignore_with_addsign.cf]: I guess this was copy/pasted. Should be `Test that @if also ignores lines starting with '@'`.

**Comment by kacf** [comment]
Great stuff!

**Comment by basvandervlies** [comment]
Just for the record. Is there a test that seed if cf-promises parse all lines and there is nothing on stdout/stderr? without this patch all '@' chars are outputted. 

thanks @kacf

File: tests/acceptance/00_basics/macros/if_ignore_with_addsign.cf (hunk 2 above)
Comment (basvandervlies) [inline tests/acceptance/00_basics/macros/if_ignore_with_addsign.cf]: Your right. I just copied a another acceptance test

File: libpromises/cf3lex.l (hunk 1 above)
Comment (tzz) [inline libpromises/cf3lex.l]: The original ate up to the next `@` so I'm not sure how the new version will work.

Comment (basvandervlies) [inline libpromises/cf3lex.l]: This is one reads the whole line instead per char but skips line starting with @ then it switches to per char read and parse all chars and seea the @endiff. When I turned on the Lex debugger I was surprised by the one char parsing when in ignore mode. Was difficult to  follow 😀

**Comment by kacf** [comment]
> Just for the record. Is there a test that seed if cf-promises parse all lines and there is nothing on stdout/stderr? without this patch all '@' chars are outputted.

No, not ATM I believe. But we have recently been talking about adding a generic check inside the `testall` script which catches unexpected text, particularly `error:` and `warning:` messages, since we have been burned by these several times. Currently the test framework only checks output in tests that explicitly look for it. I suppose that would catch this case as well.

**Comment by kacf** [comment]
trigger build

**Comment by kacf** [comment]
Thanks Bas, fixed slightly incorrect commit message (referring to some unrelated ticket) and merged manually. Cherry-picked to 3.7.x and 3.8.x as well.

**Pull request closed without merging**